src/main/java/org/wso2/lsp4intellij/editor/EditorEventManager.java

```diff
@@ -1456,19 +1475,24 @@ public void requestAndShowCodeActions() {
                         int startOffset = editor.getDocument().getLineStartOffset(line);
                         int endOffset = editor.getDocument().getLineEndOffset(line);
                         TextRange range = new TextRange(startOffset, endOffset);
-
-                        try {
-                            this.anonHolder
-                                    .newAnnotation(HighlightSeverity.INFORMATION, codeAction.getTitle())
-                                    .range(range)
-                                    .withFix(new LSPCodeActionFix(FileUtils.editorToURIString(editor), codeAction))
-                                    .create();
-
-                            SmartList<Annotation> asList = (SmartList<Annotation>) this.anonHolder;
-                            this.annotations.add(asList.get(asList.size() - 1));
-                            diagnosticSyncRequired = true;
-                        } catch (IllegalArgumentException e) {
-                            LOG.warn("Error when creating a new annotation");
+                        Tuple3<HighlightSeverity, TextRange, LSPCodeActionFix> triple =
+                                new Tuple3<>(
+                                        HighlightSeverity.INFORMATION,
+                                        range,
+                                        new LSPCodeActionFix(FileUtils.editorToURIString(editor), codeAction)
+                                );
+                        boolean found = false;
+                        for (Tuple3<HighlightSeverity, TextRange, LSPCodeActionFix> silentAnnotation : silentAnnotations) {
+                            if (silentAnnotation.getSecond().getStartOffset() == startOffset
+                                    && silentAnnotation.getSecond().getEndOffset() == endOffset
+                                    && silentAnnotation.getThird().getText().equals(codeAction.getTitle())) {
+                                found = true;
+                                break;
+                            }
+                        }
```

Comment:
You can improve here using java streams API
```suggestion
    boolean found = silentAnnotations.stream()
        .anyMatch(silentAnnotation ->
                silentAnnotation.getSecond().getStartOffset() == startOffset &&
                silentAnnotation.getSecond().getEndOffset() == endOffset &&
                silentAnnotation.getThird().getText().equals(codeAction.getTitle())
        );
```